pkg/fleet/internal/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,102 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package bootstrap provides logic to self-bootstrap the installer.
+package bootstrap
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/pkg/fleet/env"
+	"github.com/DataDog/datadog-agent/pkg/fleet/internal/exec"
+	"github.com/DataDog/datadog-agent/pkg/fleet/internal/oci"
+)
+
+const (
+	installerPackage = "datadog-installer"
+	installerBinPath = "bin/installer/installer"
+
+	rootTmpDir = "/opt/datadog-installer/tmp"
+)
+
+// Install self-installs the installer package from the given URL.
+func Install(ctx context.Context, env *env.Env, url string) error {
+	err := os.MkdirAll(rootTmpDir, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create temporary directory: %w", err)
+	}
+	tmpDir, err := os.MkdirTemp(rootTmpDir, "")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary directory: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	cmd, err := downloadInstaller(ctx, env, url, tmpDir)
+	if err != nil {
+		return fmt.Errorf("failed to download installer: %w", err)
+	}
+	return cmd.Install(ctx, url, nil)
+}
+
+// InstallExperiment self-installs the installer package from the given URL as an experiment.
+func InstallExperiment(ctx context.Context, env *env.Env, url string) error {
```

Comment:
nit: when reading I wondered if there could be a difference in the initial setup commands,
creating an intermediate `function install(ctx, env, url,  stable/experiment)` will avoid mismatches